### PR TITLE
ambient: minor refactor existsProxies/wantProxies

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

- minor refactor existsProxies/wantProxies

### CODE REVIEW

File: pilot/pkg/ambient/controller/waypoint_controller.go [hunk 1]
```diff
@@ -80,21 +83,11 @@ func NewWaypointProxyController(client kubelib.Client, clusterID cluster.ID, con
...
-			}, subresources...)
-			return err
-		},
 	}
 
 	rc.queue = controllers.NewQueue("waypoint proxy",
 		controllers.WithReconciler(rc.Reconcile),
-		controllers.WithMaxAttempts(5))
-
+		controllers.WithMaxAttempts(maxReconcileAttempts))
```
Comment (howardjohn) [inline pilot/pkg/ambient/controller/waypoint_controller.go:90]: Nit: IMO (but probably not others opinion) this refactor of moving the constant just masks readability, not improves it

File: pilot/pkg/ambient/controller/waypoint_controller.go [hunk 2]
```diff
@@ -127,7 +120,7 @@ func (rc *WaypointProxyController) Reconcile(name types.NamespacedName) error {
 		// Mostly used to avoid issues with local runs
 		return fmt.Errorf("injection config invalid, skipping reconile")
 	}
-	log := waypointLog.WithLabels("gateway", name.String())
+	scopedLog := waypointLog.WithLabels("gateway", name.String())
```
Comment (howardjohn) [inline pilot/pkg/ambient/controller/waypoint_controller.go:123]: replacing `log` pattern is intentionally used here (and elsewhere) to ensure folks use the correctly scoped log and don't accidentally fall back to the unscoped one

**Review by howardjohn — COMMENTED** [review]
Can you explain why we want to do this? https://github.com/istio/istio/blob/47af21a4088d20b3530df8872f1678f8fbb43618/pilot/pkg/config/kube/gateway/deploymentcontroller.go#L60 explains the rationale for the current approach

**Comment by zirain** [comment]
> Can you explain why we want to do this?
> 
> https://github.com/istio/istio/blob/47af21a4088d20b3530df8872f1678f8fbb43618/pilot/pkg/config/kube/gateway/deploymentcontroller.go#L60
> 
> explains the rationale for the current approach

my original just like `no one likes YAML templating...`, but I do not notice the comments (that convince me).
I will revert this part.

**Pull request closed without merging**